tools/docker/demo/Dockerfile

```diff
@@ -1,40 +1,68 @@
-FROM ubuntu:18.04
+FROM ubuntu:20.04
 
-ARG SPARK_VERSION=3.2.0
+ARG MMLSPARK_VERSION=0.9.5
+ARG DEBIAN_FRONTEND=noninteractive
+
+ENV SPARK_VERSION=3.2.1
+ENV HADOOP_VERSION=2.7
+ENV MMLSPARK_VERSION=${MMLSPARK_VERSION}
+ENV JAVA_HOME /usr/lib/jvm/java-1.11.0-openjdk-amd64
+
+# Install required packages
 RUN apt-get -qq update \
-    && apt-get -qq upgrade \
-    && apt-get -qq -y install curl bzip2 \
-    && curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o /tmp/miniconda.sh \
+    && apt-get -qq -y install \
+      curl \
+      bzip2 \
+      wget \
+      default-jre \
+      openmpi-bin \
+    && apt-get -qq -y autoremove \
+    && apt-get autoclean \
+    && rm -rf /var/lib/apt/lists/* /var/log/dpkg.log
+
+# Install conda and required dependencies
+RUN curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o /tmp/miniconda.sh \
     && bash /tmp/miniconda.sh -bfp /usr/local \
     && rm -rf /tmp/miniconda.sh \
-    && conda install -y python=3 \
     && conda update conda \
-    && apt-get install openjdk-8-jre -y \
-	&& conda install -c conda-forge pyspark=${SPARK_VERSION} \
-    && apt-get -qq -y remove curl bzip2 \
-    && apt-get -qq -y autoremove \
-    && apt-get autoclean \
-    && rm -rf /var/lib/apt/lists/* /var/log/dpkg.log \
+    && conda install -y python=3 jupyter pyspark \
     && conda clean --all --yes
 
 ENV PATH /opt/conda/bin:$PATH
-ENV JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk-amd64
 
-RUN conda install jupyter \
-    && jupyter-notebook --generate-config \
+# Download and install Spark
+RUN wget https://downloads.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz \
+    && tar -xzf spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz \
+    && mv spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION} /opt/spark \
+    && rm spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz
+
+ENV SPARK_HOME /opt/spark
+ENV PYTHONPATH $SPARK_HOME/python/:$SPARK_HOME/python/lib/py4j*:$PYTHON_PATH
+ENV PATH $SPARK_HOME/bin/:$SPARK_HOME/python/:$PATH
+
+RUN apt-get remove --purge -y \
```

Comment:
I think this needs to be run when apt-get install happens otherwise it will only increase the image size

Reply:
I think you might be talking about `apt-get autoclean / autoremove`. This remove sub-command is to remove packages we installed in the first run on Line 13. Let me know if I am wrong here.